addons/survey/tests/common.py

```diff
@@ -314,3 +314,80 @@ def setUp(self):
             'sequence': 3,
             'question_type': 'numerical_box',
         })
+
+
+class TestSurveyResultsCommon(SurveyCase):
+    def setUp(self):
+        super(TestSurveyResultsCommon, self).setUp()
+        self.survey_manager = mail_new_test_user(
+            self.env, name='Gustave Doré', login='survey_manager', email='survey.manager@example.com',
+            groups='survey.group_survey_manager,base.group_user'
+        )
+
+        # Create survey with questions
+        self.survey = self.env['survey.survey'].create({
+            'title': 'Test Survey Results',
+            'questions_layout': 'one_page'
+        })
+        self.question_cb = self._add_question(
+            None, 'What is your name', 'char_box', survey_id=self.survey.id, sequence='1')
+        self.question_num = self._add_question(
+            None, 'What is your age', 'numerical_box', survey_id=self.survey.id, sequence='2')
+        self.question_sc = self._add_question(
+            None, 'Are you a cat or a dog person', 'simple_choice', survey_id=self.survey.id,
+            sequence='3', labels=[{'value': 'Cat'},
+                                    {'value': 'Dog'}])
+        self.question_mc = self._add_question(
+            None, 'What do you like most in our tarte al djotte', 'multiple_choice', survey_id=self.survey.id,
+            sequence='4', labels=[{'value': 'The gras'},
+                                    {'value': 'The bette'},
+                                    {'value': 'The tout'},
+                                    {'value': 'The regime is fucked up'}])
+        self.question_mx1 = self._add_question(
+            None, 'When do you harvest those fruits', 'matrix', survey_id=self.survey.id, sequence='5',
+            labels=[{'value': 'Spring'}, {'value': 'Summer'}],
+            labels_2=[{'value': 'Apples'},
+                        {'value': 'Strawberries'}])
+        self.question_mx2 = self._add_question(
+            None, 'How often should you water those plants', 'matrix', survey_id=self.survey.id, sequence='6',
+            labels=[{'value': 'Once a month'}, {'value': 'Once a week'}],
+            labels_2=[{'value': 'Cactus'},
+                        {'value': 'Ficus'}])
+
+        # Question answers ids
+        self.cat_id = self.question_sc.suggested_answer_ids[0].id
+        self.dog_id = self.question_sc.suggested_answer_ids[1].id
```

Comment:
You should be able to shorten it a bit:

`[self.cat_id, self.dog_id] = self.question_sc.suggested_answer_ids.ids`